File: 1342/main.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int numberOfSteps(int num) {
        return numberOfStepsRecursive(num, 0);
    }

    int numberOfStepsRecursive(int num, int currentSteps) {
        if (num == 0) {
            return currentSteps;
        }

        if (num % 2 == 1) {
            return numberOfStepsRecursive(num - 1, currentSteps + 1);
        } else {
            return numberOfStepsRecursive(num / 2, currentSteps + 1);
        }
    }
};

int main() {
    Solution solution;
    cout << solution.numberOfSteps(14) << endl;
    cout << solution.numberOfSteps(8) << endl;
}

File: 1342/main.rs
fn main() {
    println!("{:?}", Solution::number_of_steps(8));
    println!("{:?}", Solution::number_of_steps(14));
}

struct Solution;

impl Solution {
    pub fn number_of_steps(num: i32) -> i32 {
        let mut decrementor = num;
        let mut counter = 0;
        while decrementor != 0 {
            match decrementor % 2 {
                0 => decrementor /= 2,
                1 => decrementor -= 1,
                _ => ()
            }
            counter += 1;
        }
        return counter;
    }
}
